service/src/main/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderService.java

```diff
@@ -256,6 +256,16 @@ public void removeOrderProperties(Map<String, Object> caseData) {
         Arrays.stream(InterimOrderKey.values()).forEach(ccdField -> caseData.remove(ccdField.getKey()));
     }
 
+    public JudgeAndLegalAdvisor getAllocatedJudgeFromMostRecentOrder(CaseData caseData) {
+        Optional<Element<GeneratedOrder>> generatedOrder = caseData.getOrderCollection()
+            .stream().reduce((first, last) -> last);
+
+        if (generatedOrder.isPresent()) {
+            return generatedOrder.get().getValue().getJudgeAndLegalAdvisor();
+        }
+        return JudgeAndLegalAdvisor.builder().build();
```

Comment:
minor: but how about
```java
return generatedOrder
            .map(Element::getValue)
            .map(GeneratedOrder::getJudgeAndLegalAdvisor)
            .map(JudgeAndLegalAdvisor::getJudgeEmailAddress)
            .orElse("");
```
This way we can still return the supplied email address or an empty string as the case may be.

Reply:
I'm returning the entire JudgeAndLegalAdvisorObject so can't do exactly like you suggested. But agree it looks much nicer like above so i've changed to be like this.. 

    public JudgeAndLegalAdvisor getAllocatedJudgeFromMostRecentOrder(CaseData caseData) {
        Optional<Element<GeneratedOrder>> generatedOrder = caseData.getOrderCollection()
            .stream().reduce((first, last) -> last);
        
        return generatedOrder
            .map(Element::getValue)
            .map(GeneratedOrder::getJudgeAndLegalAdvisor)
            .orElse(JudgeAndLegalAdvisor.builder().build());
    }

@Gogetter ?

Reply:
yes, this works